### PR TITLE
Swaps keyboard dropdown

### PR DESCRIPTION
**Pull request by darkwing**
Explanation:  

At present the dropdowns used in swaps are not accessible via keyboard.  This PR makes it so that:
- Users can tab to the dropdown, press Enter, and then press tab to get from item to item, then press Enter to select one
- Pressing Escape closes the dropdown

Manual testing steps:  
  - Open each dropdown via Enter key, press Tab to move from item to item
  - Press Enter to select items
  - Open each dropdown via Enter key, press Escape to close the dropdown

Would be good for @danjm to review as I'd like to ensure everything looks good on Linux.

Keyboard

### CODE REVIEW

File: ui/app/pages/swaps/dropdown-search-list/dropdown-search-list.js [hunk 1]
```diff
@@ -29,12 +29,15 @@ export default function DropdownSearchList ({
...
+    onClose && onClose()
+  }
   const onClickItem = useCallback((item) => {
     onSelect && onSelect(item)
     setSelectedItem(item)
-    setIsOpen(false)
-    onClose && onClose()
-  }, [onClose, onSelect])
+    close()
+  }, [onSelect])
```
Comment (brad-decker) [inline ui/app/pages/swaps/dropdown-search-list/dropdown-search-list.js:40]:
```suggestion
  }, [close, onSelect])
```

without close being in the dependency array, `useCallback` would only get called anytime onSelect changes, which could result in holding onto a reference to the close function from previous renders (and also anything that also uses that reference to the close function).

**Review by brad-decker — CHANGES_REQUESTED** [review]
LGTM, just one suggested update

**Comment by darkwing** [comment]
This should be good to go.

**Review by brad-decker — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**